lib/handler.js

```diff
@@ -73,52 +73,58 @@ class Handler {
-    let validators = []
-    let excludes = (settings.exclude)
-      ? settings.exclude.split(',').map(val => val.trim()) : []
-    let includes = [ 'approvals', 'title', 'label', 'milestone', 'description', 'projects', 'assignee', 'dependent' ]
-      .filter(validator => excludes.indexOf(validator) === -1)
-
-    includes.forEach(validator => {
-      validators.push(require(`../lib/${validator}`)(pullRequest, context, settings))
-    })
-
-    console.info(config)
-    return Promise.all(validators).then(results => {
-      let failures = results.filter(validated => !validated.mergeable)
-
-      let status, description
-      if (failures.length === 0) {
-        status = 'success'
-        description = 'Okay to merge.'
-      } else {
-        status = 'failure'
-        description = `## Mergeable has found the following failed checks`
-        for (let validated of failures) {
-          description += `\n - ${validated.description}`
+    try {
+      var config = await Configuration.instanceWithContext(context)
+      let settings = config.settings.mergeable.pull_requests || config.settings.mergeable
+      let validators = []
+      let excludes = (settings.exclude)
+        ? settings.exclude.split(',').map(val => val.trim()) : []
+      let includes = [ 'approvals', 'title', 'label', 'milestone', 'description', 'projects', 'assignee', 'dependent' ]
+        .filter(validator => excludes.indexOf(validator) === -1)
+
+      includes.forEach(validator => {
+        validators.push(require(`../lib/${validator}`)(pullRequest, context, settings))
+      })
+      console.info(config)
+      return Promise.all(validators).then(results => {
+        let failures = results.filter(validated => !validated.mergeable)
+
+        let status, description
+        if (failures.length === 0) {
+          status = 'success'
+          description = 'Okay to merge.'
+        } else {
+          status = 'failure'
+          description = `## Mergeable has found the following failed checks`
+          for (let validated of failures) {
+            description += `\n - ${validated.description}`
+          }
+          description += `\n ___ \n **Please address the problems found above!**`
         }
-        description += `\n ___ \n **Please address the problems found above!**`
-      }
 
+        checks.update(
+          context,
+          checkRunResult.data.id,
+          'Mergeable',
+          'completed',
+          status,
+          {
+            title: `Result: ${status}`,
+            summary: description})
+
+        console.info({state: status, description: description})
+      })
+    } catch (err) {
       checks.update(
         context,
         checkRunResult.data.id,
         'Mergeable',
         'completed',
-        status,
+        'action_required',
         {
-          title: `Result: ${status}`,
-          summary: description})
-
-      console.info({state: status, description: description})
-    }).catch(error => {
```

Comment:
This catches `promise` rejections. Removing this will cause unknown errors to go unlogged. I really don't think it should be removed. Normal try catches will do anything here as this is async.

Furthermore at this point it's errors thrown by validators are not due to bad configuration and are typically errors caused by system bugs snda hence the message should tell the user that a "system error has occurred"  -- it's not something they can do on their end. I also really would like these errors to be logged.